File: ulyp-ui/src/main/java/com/ulyp/ui/CallTreeNodeContent.java
package com.ulyp.ui;

import com.ulyp.core.CallRecord;
import javafx.geometry.Pos;
import javafx.scene.layout.StackPane;

// TODO should be a better name
public class CallTreeNodeContent extends StackPane {

    public CallTreeNodeContent(CallRecord node, RenderSettings renderSettings, long totalNodeCountInTree) {

        setAlignment(Pos.CENTER_LEFT);
        getChildren().addAll(new CallRecordTreeNodeRelativeWeight(node, totalNodeCountInTree), new RenderedCallRecord(node, renderSettings));
    }
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/CallTreeNodeContent.kt
package com.ulyp.ui

import com.ulyp.core.CallRecord
import javafx.geometry.Pos
import javafx.scene.layout.StackPane

// TODO should be a better name
class CallTreeNodeContent(node: CallRecord, renderSettings: RenderSettings?, totalNodeCountInTree: Long) :
    StackPane() {
    init {
        alignment = Pos.CENTER_LEFT
        children.addAll(
            CallRecordTreeNodeRelativeWeight(node, totalNodeCountInTree),
            RenderedCallRecord(node, renderSettings!!)
        )
    }
}